pkg/filesystem/path/builder.go

```diff
@@ -268,3 +336,10 @@ func (rp builderRelativeParser) ParseFirstComponent(componentWalker ComponentWal
 	}
 	return r, nil, nil
 }
+
+func validateWindowsPathComponent(component string) error {
+	if strings.ContainsFunc(component, unicode.IsControl) || strings.ContainsAny(component, "<>:\"/\\|?*") {
+		return status.Errorf(codes.InvalidArgument, "Path component contains a control character or invalid character %#v.", component)
```

Comment:
Please remove the trailing dot. Also move the `%#v` to the right place in the sentence.
```go
return status.Errorf(codes.InvalidArgument, "Path component %#v contains a control character or invalid character %#v", component)
```
That said, the general pattern in the Buildbarn codebase is to never format errors with values that also happen to be arguments of the function. Probably better to do this?
```go
if strings.ContainsFunc(component, unicode.IsControl) {
    return status.Error(codes.InvalidArgument, "Path component contains control characters")
}
if strings.ContainsAny(component, "<>:\"/\\|?*") {
   return status.Error(codes.InvalidArgument, "Path component contains reserved characters")
}
```
And then let the caller do a `util.StatusWrapf(err, "Invalid path component %#v", component.String())`.

Reply:
Note that there are also some additional restrictions on Windows, such as filenames not being allowed to end with dots or whitespace. And there are obviously reserved names such as LPT1. But whatever. I think we should only add checks for that if it ever becomes problematic. This should do.

Reply:
Good point on the former! I did not make that connection, just thought about the error myopically. That is a great rule.

Argh windows... "But wait, there is more!"